src/Symfony/Component/Validator/Constraints/NullValidator.php

```diff
@@ -11,25 +11,12 @@
 
 namespace Symfony\Component\Validator\Constraints;
 
-use Symfony\Component\Validator\Constraint;
-use Symfony\Component\Validator\ConstraintValidator;
-
 /**
  * @author Bernhard Schussek <bschussek@gmail.com>
  *
  * @api
+ *
+ * @deprecated Deprecated since version 2.3.28, to be removed in 3.0. Use
```

Comment:
I just talked with @fabpot in person and yes the deprecation and trigger_error is only in 2.7.

2.3 will be PHP 7 compatible for Symfony for out-of-the-box, but we won't prevent others from still using the old constraints, etc. (and such make it PHP 7 incompatible)

That way 2.3 is compatible out-of-the-box for Drupal + PHP 7, but you could still shoot yourself in the foot with a custom extension.

In 2.7 then it will start with deprecation notices.